sns/parity_math.c

```diff
@@ -139,57 +347,95 @@ static void forward_rectification(struct m0_sns_ir *ir,
  * for computing failed parity.
  */
 static void failed_data_blocks_xform(struct m0_sns_ir *ir);
+#endif /* !ISAL_ENCODE_ENABLED */
+
 static inline bool is_valid_block_idx(const  struct m0_sns_ir *ir,
 				      uint32_t block_idx);
 
+#if !ISAL_ENCODE_ENABLED
 static bool is_data(const struct m0_sns_ir *ir, uint32_t index);
-
+#endif /* !ISAL_ENCODE_ENABLED */
 
 static bool is_usable(const struct m0_sns_ir *ir,
 		      const struct m0_bitmap *in_bmap,
 		      struct m0_sns_ir_block *failed_block);
 
 static uint32_t last_usable_block_id(const struct m0_sns_ir *ir,
 				     uint32_t block_idx);
+
+#if !ISAL_ENCODE_ENABLED
 static inline  bool are_failures_mixed(const struct m0_sns_ir *ir);
 
 static inline const struct m0_matrix* recovery_mat_get(const struct m0_sns_ir
 						       *ir,
 						       uint32_t failed_idx);
+#endif /* !ISAL_ENCODE_ENABLED */
+
+static inline uint32_t ir_blocks_count(const struct m0_sns_ir *ir);
 
-static inline uint32_t block_count(const struct m0_sns_ir *ir);
+#if ISAL_ENCODE_ENABLED
+/**
+ * Initialize fields required for incremental recovery using Intel ISA.
+ * @param[in] ir       - Pointer to incremental recovery structure.
+ * @retval    0          Success
+ * @retval    -ENOMEM    Failed to allocate memory
+ */
+static int isal_ir_init(struct m0_sns_ir *ir);
 
-static void (*calculate[M0_PARITY_CAL_ALGO_NR])(struct m0_parity_math *math,
-						const struct m0_buf *data,
-						struct m0_buf *parity) = {
+/**
+ * Free fields initialized for incremental recovery using Intel ISA.
+ * @param[in] ir - pointer to incremental recovery structure.
+ */
+static void isal_ir_fini(struct m0_sns_ir *ir);
+#endif /* ISAL_ENCODE_ENABLED */
+
+static int (*calculate[M0_PARITY_CAL_ALGO_NR])(struct m0_parity_math *math,
+					       const struct m0_buf *data,
+					       struct m0_buf *parity) = {
```

Comment:
How ->calculate() function can fail? All its input and output buffers are preallocated. The same question for other s/void/int/ changes in return types.

Reply:
Reverted the change.